main/wallet-backend.js

```diff
@@ -69,119 +83,160 @@ class WalletBackend {
   }
 
   /**
-   * @param {string} from
    * @param {string} to
-   * @param {FilecoinNumber} amount
+   * @param {ethers.BigNumber} amount
+   * @param {function} fn
    * @returns {Promise<string>}
    */
-  async transferFunds (from, to, amount) {
-    assert(this.provider)
+  async runTransaction (to, amount, fn) {
+    assert(this.signer)
 
     /** @type {FILTransactionProcessing} */
     const transaction = {
       timestamp: new Date().getTime(),
       status: 'processing',
       outgoing: true,
-      amount: amount.toString(),
-      address: to
+      amount: ethers.utils.formatUnits(amount, 18),
+      address: to.startsWith('f1')
+        ? to
+        : delegatedFromEthAddress(to, CoinType.MAIN)
     }
     this.transactions.push(transaction)
     this.onTransactionUpdate()
 
     try {
-      const gasLimit = await this.getGasLimit(from, to, amount)
-      // Ensure transaction has enough gas to succeed
-      const gasOffset = new FilecoinNumber('100', 'attofil')
-      const amountMinusGas = amount.minus(gasLimit).minus(gasOffset)
-      const message = new Message({
-        to,
-        from,
-        nonce: await this.provider.getNonce(from),
-        value: amountMinusGas.toAttoFil(),
-        method: 0,
-        params: ''
-      })
-      const messageWithGas = await this.provider.gasEstimateMessageGas(
-        message.toLotusType()
-      )
-      const lotusMessage = messageWithGas.toLotusType()
-      const msgValid = await this.provider.simulateMessage(lotusMessage)
-      assert(msgValid, 'Message is invalid')
-      const signedMessage = await this.provider.wallet.sign(from, lotusMessage)
-      const { '/': cid } = await this.provider.sendMessage(signedMessage)
-
+      const tx = await fn()
+      console.log({ tx })
+      assert(tx.hash, 'Transaction hash not found')
+      const cid = await this.convertEthTxHashToCid(tx.hash)
+      console.log({ cid })
       transaction.hash = cid
       this.onTransactionUpdate()
-
       return cid
     } catch (err) {
       console.error(err)
       transaction.status = 'failed'
       this.onTransactionUpdate()
-
       throw err
     }
   }
 
+  /**
+   * @param {string} hash
+   * @returns {Promise<string>}
+   */
+  async convertEthTxHashToCid (hash) {
+    console.log('convertEthTxHashToCid', { hash })
+    for (let i = 0; i < 10; i++) {
+      const res = await fetch('https://graph.glif.link/query', {
+        headers: {
+          accept: '*/*',
+          'accept-language': 'en-US,en;q=0.9,de;q=0.8',
+          'content-type': 'application/json'
+        },
+        body: JSON.stringify({
+          operationName: 'Message',
+          variables: {
+            cid: hash
+          },
+          query: 'query Message($cid: String!) {message(cid: $cid) {cid}}'
+        }),
+        method: 'POST'
+      })
+      const body = await res.json()
+      if (body.data.message) {
+        return body.data.message.cid
+      }
+      console.log(body.errors)
```

Comment:
Do we have Sentry integration available here? Should we report the error to Sentry in such case?

Reply:
Errors here are expected, when the transaction can't yet be found. These aren't critical, only debug errors

Reply:
I've changed to only log if we see an error here that we're not expecting